stdlib/bytes.ml

```diff
@@ -207,6 +207,51 @@ let mapi f s =
     r
   end
 
+let fold f s accu =
+  let l = length s in
+  let rec loop i acc =
+    if i < l then
+      loop (i+1) (f (unsafe_get s i) acc)
+    else
+      acc
+  in
+  loop 0 accu
+
+exception Abort
+
+let for_all f s =
```

Comment:
I'd suggest to implement for_all and exists as in array.ml, i.e. using a recursive loop (except if some micro-benchmark shows this is really slower, in which case I would suggest a while-loop for both).  (In general, when exception are used like that locally for control flow, it is a good idea to use raise_notrace for performance reason.)